CONTRIBUTING.md

```diff
@@ -1,10 +1,63 @@
-You want to contribute? We really appreciate that.
-Please have a look at this list to facilitate your contribution:
+# Want to contribute some code? 
 
-* Before you start coding, create a Github issue at https://github.com/jamulussoftware/jamulus/issues. In that new issue write your specification and that you are planning to do the coding. Then we can discuss the specification. Only start coding after we agreed to a specification.
-* If the Github issue for your feature/bug fix already exists, then write a message in that issue that you want to work on it.
-* Respect the existing code style: Tab size=4, insert spaces.
-* Make sure (if possible) that your code compiles on Windows/Mac/Linux.
-* Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
+## We’d really appreciate your support!
 
-Look here for other further guidelines: https://github.com/jamulussoftware/jamulus/discussions/915
+- If the [Github issue](https://github.com/jamulussoftware/jamulus/issues) for your feature/bug fix already exists, write a message in that issue saying you want to work on it. 
+- Otherwise, please [post on the GitHub Discussions](https://github.com/jamulussoftware/jamulus/discussions) and say that you are planning to do some coding and explain why. Then we can discuss the specification. 
+- Please only start coding after we have agreed to a specification to avoid putting lots of work into something which may not get accepted later.
+
+## Jamulus project/source code general principles
+
+### Stability
+This has been, and must continue to be, the most important requirement. It is particularly important for use cases such as the Worldjam where instability during a live performance would be unacceptable. The following two principles are designed to support this principle.
+
+### [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
+If a feature or function can be achieved in another way by another system or method, then that is preferable to building that feature into Jamulus. 
+
+### Source code consistency
+- Respect the existing code style: Tab size=4, insert spaces.
+- Space before and after `(` and `)`, except no space between `)` and `;`, and no space before an empty `()`.
+- All bodies of `if`, `else`, `while`, `for`, etc., to be enclosed in braces `{` and `}`, on separate lines.
+
+Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
+
+Please check your Github fork. It should build your branch for all the supported platforms (though not all supported platform versions) when you push. Make sure there are no errors. If there are, resolve them before creating a pull request.
```

Comment:
Are we sure that this is possible?
I suspect the autobuild only runs for release tags, on master and on PRs. This would mean that no autobuild runs on your own fork's feature branches.

Reply:
Volker wrote that, but I guess it's possible he never tried it?

Reply:
I just checked: As I suspected, there is no autobuild for this case.

Here's a feature branch of mine in my fork: https://github.com/hoffie/jamulus/tree/recorder-cleanup
The `Actions` tab does not list any autobuild jobs within the last hours.

The PR (#1463) has the proper CI jobs.

```suggestion
Please run a local build test. Make sure there are no errors. After opening a pull request, keep an eye on the CI checks for quality or compile issues and fix them as required.
```

As an alternative, we could change the autobuild logic (@ann0see @nefarius2001), if that's wanted.

Reply:
I think every push should trigger a build.  (It can quite happily kill older builds on the branch, too, if that's feasible.)